site/src/testHelpers/storybook.tsx

```diff
@@ -45,25 +45,39 @@ export const withDashboardProvider = (
   );
 };
 
+type MessageEvent = Record<"data", string>;
+type CallbackFn = (ev?: MessageEvent) => void;
+
 export const withWebSocket = (Story: FC, { parameters }: StoryContext) => {
-  if (!parameters.webSocket) {
-    console.warn(
-      "Looks like you forgot to add websocket messages to the story",
-    );
+  const events = parameters.webSocket;
+
+  if (!events) {
+    console.warn("Your forgot to add the `parameters.webSocket` to your story");
+    return <Story />;
   }
 
+  const listeners = new Map<string, CallbackFn>();
+  let callEventsDelay: number;
+
   // @ts-expect-error -- TS doesn't know about the global WebSocket
   window.WebSocket = function () {
     return {
-      addEventListener: (
-        type: string,
-        callback: (ev: Record<"data", string>) => void,
-      ) => {
-        if (type === "message") {
-          parameters.webSocket?.messages.forEach((message) => {
-            callback({ data: message });
-          });
-        }
+      addEventListener: (type: string, callback: CallbackFn) => {
+        listeners.set(type, callback);
+
+        // Runs when the last event listener is added
+        clearTimeout(callEventsDelay);
+        callEventsDelay = window.setTimeout(() => {
```

Comment:
Out of curiosity, why do we need the timeout?  I think it makes sense since it allows us to make sure all the listeners are hooked up first but was there a bug with how it was before?

Reply:
Yes, the problem with the previous implementation is that it did not respect the order of the events so if the `error` listener was defined before it would call it first even if it was the third event in the list for example.

Reply:
Ohhhhh I see!  Makes sense.